### ISSUE TITLE
"Docker extension recommended for this file type" with only a Dart file open in remote session

### ISSUE THREAD

**Issue by DanTup** [microsoft/vscode#84039]
I'm using today's Insiders and have a Flutter project open in a remote Docker container with a single Dart file open:

<img width="1059" alt="Screenshot 2019-11-06 at 9 38 54 am" src="https://user-images.githubusercontent.com/1078012/68286602-4496d580-0079-11ea-9e45-64f7acc24b6d.png">

When I opened it, I got an unexpected recommendation for the Docker extension. There is certainly a Dockerfile in the `.devcontainer` folder, but it's not open.

If I run the **Reload Window** then the recommendation reappears:

<img width="1059" alt="Screenshot 2019-11-06 at 9 39 57 am" src="https://user-images.githubusercontent.com/1078012/68286688-6b550c00-0079-11ea-93ae-01eb116abd22.png">

It's not clear whether VS Code has the notification is being triggered by the Dockerfile in the devcontainer folder, or whether VS Code has decided the Docker extension is good for editing Dart (I'm not sure how to check that).

**Comment by DanTup** [comment]
Seems unrelated to the Dart file - it pops up whenever I open a remote session even when there are zero files open.

**Comment by sandy081** [comment]
This is because the docker file is being opened by the extension. @chrmarti is this correct?

**Comment by chrmarti** [comment]
We don't open the Dockerfile. Wouldn't it show as an editor tab if it were open?

**Comment by sandy081** [comment]
I see a model is being created for the Docker file. It need not be necessary to open in the editor.

**Comment by DanTup** [comment]
> I see a model is being created for the Docker file. It need not be necessary to open in the editor.

It seems strange to prompt a user with a recommendation for a file that they don't see open in the editor (assuming that's the cause). Shouldn't it only trigger for visible editors?

**Comment by DanTup** [comment]
(this is currently tagged as needs-more-info, but I'm not sure what else I can provide?)

**Comment by sandy081** [comment]
> It seems strange to prompt a user with a recommendation for a file that they don't see open in the editor (assuming that's the cause). Shouldn't it only trigger for visible editors?

True. But it has been like this from beginning and can be some recursions if changed. 

Still not sure who is opening the docker file in the dev container mode. 

@chrmarti This is happening only in dev container mode, so this change can be coming from the extension. Can you please confirm?

**Comment by DanTup** [comment]
> True. But it has been like this from beginning and can be some recursions if changed.

Ok, but surely it's the best fix here? Even if you track down who is opening it and stop it, it doesn't make sense it other extensions "open" files without them being visible and users get notifications for recommendations. It would be better to fix the behaviour to do what makes sense - who it when an editor is visible for a while, not when an extension may be doing something internal? (for ex. if you want to map offsets to line/columns you may need to "open" a document).

**Comment by chrmarti** [comment]
@sandy081 You're right, Remote-Containers reads the Dockerfile as a workspace document. I don't have to do it that way, it was just convenient to do so.

**Comment by sandy081** [comment]
I would not go and do a big change to fix this. @chrmarti Is it possible to not to read as a workspace document?

**Comment by chrmarti** [comment]
I can look into it.
